From 707e2ef267f5798245b5d0d646aaa121137fdd43 Mon Sep 17 00:00:00 2001
From: Michal Klocek <michal.klocek@qt.io>
Date: Thu, 22 Aug 2024 09:54:45 +0200
Subject: [PATCH] Fix cross comile error for apalis-imx8

Deals with:

"error: operands to '?:' have different types 'intptr_t' {aka 'long
int'} and 'EGLNativeDisplayType' {aka 'wl_display*'}"

Task-number: QTBUG-126655
Change-Id: I62b2369b2dff2b7a2929a91d59cde4d24b8c2398
Reviewed-by: Allan Sandfeld Jensen <allan.jensen@qt.io>
(cherry picked from commit 639d9f5b547486281bb1c218dc87f968c69a2642)
Reviewed-by: Qt Cherry-pick Bot <cherrypick_bot@qt-project.org>
Upstream-Status: Accepted
---
 src/core/ozone/gl_ozone_egl_qt.cpp | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/core/ozone/gl_ozone_egl_qt.cpp b/src/core/ozone/gl_ozone_egl_qt.cpp
index 26d11df31..8cc793841 100644
--- a/src/core/ozone/gl_ozone_egl_qt.cpp
+++ b/src/core/ozone/gl_ozone_egl_qt.cpp
@@ -75,7 +75,8 @@ scoped_refptr<gl::GLSurface> GLOzoneEGLQt::CreateOffscreenGLSurface(gl::GLDispla
 gl::EGLDisplayPlatform GLOzoneEGLQt::GetNativeDisplay()
 {
     static void *display = GLContextHelper::getNativeDisplay();
-    static gl::EGLDisplayPlatform platform(display ? reinterpret_cast<intptr_t>(display) : EGL_DEFAULT_DISPLAY);
+    static gl::EGLDisplayPlatform platform(display ? reinterpret_cast<EGLNativeDisplayType>(display)
+                                                   : EGL_DEFAULT_DISPLAY);
     return platform;
 }

